src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala

```diff
@@ -559,20 +559,22 @@ trait NamesDefaults { self: Analyzer =>
   def removeNames(typer: Typer)(args: List[Tree], params: List[Symbol]): (List[Tree], Array[Int]) = {
     implicit val context0 = typer.context
     def matchesName(param: Symbol, name: Name, argIndex: Int) = {
-      def warn(w: String) = context0.deprecationWarning(args(argIndex).pos, param, w)
+      def warn(msg: String, since: String) = context0.deprecationWarning(args(argIndex).pos, param, msg, since)
       def checkDeprecation(anonOK: Boolean) =
         when (param.deprecatedParamName) {
           case Some(`name`)      => true
           case Some(nme.NO_NAME) => anonOK
         }
+      def version = param.deprecatedParamVersion.getOrElse("")
+      def since   = if (version.isEmpty) version else s" (since $version)"
       def checkName = {
         val res = param.name == name
-        if (res && checkDeprecation(true)) warn(s"naming parameter $name has been deprecated.")
+        if (res && checkDeprecation(true)) warn(s"naming parameter $name is deprecated$since.", version)
         res
       }
       def checkAltName = {
         val res = checkDeprecation(false)
-        if (res) warn(s"the parameter name $name has been deprecated. Use ${param.name} instead.")
+        if (res) warn(s"the parameter name $name is deprecated$since: Use ${param.name} instead", version)
```

Comment:
i was looking at the compiler-generated message, these seem to start lowercase, for example https://github.com/scala/scala/pull/5076/files#diff-5340a538b6fc701f10852f5e589bb6e3L1 or https://github.com/scala/scala/pull/5076/files#diff-faa24cfcca41cdc68f60aa788d6d2558L1